### PR TITLE
Helm local dev

### PR DESCRIPTION
**Pull request by ldelossa**
This PR implements a kubernetes driven local development environment. It's expected that kubernetes clusters be ran via minikube or docker desktop. If the developer decides to run kubernetes by a different manner they will just need to make sure their docker cli tool is configured to push build context to said kubernetes cluster. 

A makefile is used to drive the local development environment. On first run of `make deploy-local` both a clair and a postgres instance are built and ran on the local kubernetes cluster. Subsequent calls to `make deploy-local` build a new container with the latest code in the working directory and only deploys clair, leaving any writes to the database intact.

Running `make teardown-local` will remove both the cliar and postgres instances from the local kubernetes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +44,20 @@ Thus, the project was named `Clair` after the French term which translates to *c
...
+
+To deploy the local dev environment first ensure you have helm initialized on your kubernetes instance. Usually this simply involves running `helm init` as long as your kubectl is pointed to the right context. Next run `make deploy-local`. This target builds the container `clair-local:latest` using the docker cli and then deploys a postgres helm chart and a clair helm chart which runs the `clair-local:latest` tag. You may run this command idempotently to deploy the latest code in your working directory.
+
+To teardown both the postgres database and the clair instance run `make teardown-local`.
+
+Caveats:
+You may see errors which do not necessarily mean there is anything wrong when running make targets. This is due to the fact that the Makefile targets are designed to be indempotent. Reference the makefile for lines prefixed with '-' to understand which commands are allowed to fail.
+
```
Comment (jzelinskie) [inline README.md:48]: Let's make a separate doc for this that gets linked to by its own bullet point in the "getting started" section of the README.

Keeping the README as simple as possible has proven very valuable in the past.

**Review by jzelinskie — COMMENTED** [review]
Looks pretty good so far.

I actually wouldn't mind a commit that deletes the whole `contrib` directory in this PR.

File: README.md (hunk 1 above)
Comment (ldelossa) [reply README.md:48]: totally into that :+1:.

**Comment by ldelossa** [comment]
@jzelinskie if we remove the contrib dir all together, do we want to preserve anything in there like the grafana templates? We could potentially make them part of the local dev env also.

File: Documentation/local-development.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Local Development
+
+# Requirements
+* Docker cli
+* Kubernetes (minikube or docker desktop)
+* Helm 
+
+# Overview 
+The local development environment is driven by a Makefile in the root of this repository. This Makefile is responsible for building a clair container on the local kubernetes instances and deploying said container. In order to build the container the Docker cli tools are expected to be present. In order to deploy the container the Helm cli tool is expected to be present and the local kubernetes cluster **must** be running tiller.
```
Comment (jzelinskie) [inline Documentation/local-development.md]: One sentence per line.

s/clair/Clair
s/kubernetes/Kubernetes
s/cli/CLI
s/tiller/Tiller

File: Documentation/local-development.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Local Development
+
+# Requirements
+* Docker cli
+* Kubernetes (minikube or docker desktop)
+* Helm 
+
+# Overview
```
Comment (jzelinskie) [inline Documentation/local-development.md:9]: new line after headers

File: README.md [hunk 4]
```diff
@@ -43,6 +43,7 @@ Thus, the project was named `Clair` after the French term which translates to *c
 [integrations]: /Documentation/integrations.md
 [CoreOS website]: https://coreos.com/clair/docs/latest/
 [Documentation directory]: /Documentation
+[Local Development]: /Documentation/local-development.md
```
Comment (jzelinskie) [inline README.md:47]: Add a bullet to the list that actually uses this link, haha

**Review by jzelinskie — CHANGES_REQUESTED** [review]

File: Documentation/local-development.md (hunk 3 above)
Comment (ldelossa) [reply Documentation/local-development.md:9]: done.

File: Documentation/local-development.md (hunk 2 above)
Comment (ldelossa) [reply Documentation/local-development.md]: done.

File: README.md (hunk 4 above)
Comment (ldelossa) [reply README.md:47]: done.

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**